File: Leetcode/0071 Simplify Path.py
class Solution:
    def simplifyPath(self, path) -> str:
      stack = []
      for dir in path.split("/"):
        if dir == '..' and stack:
          stack.pop()
        elif dir not in ['', '.', '..']:
          stack.append(dir)
      return f'/{"/".join(stack)}'
    

File: Leetcode/0071 Simplify Path.ts
const ignore = ["", ".", ".."]

function simplifyPath(path: string): string {
    const stack: string[] = [];
    const dirs = path.split("/");
    for (const dir of dirs) {
        if(dir === ".." && stack.length > 0) {
            stack.pop();
        } else if(!ignore.includes(dir)){
            stack.push(dir);
        }
    }
    return `/${stack.join("/")}`;
}
